src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyExecutionManager.cs

```diff
@@ -189,10 +187,10 @@ public void Close()
 
                 // Dispose concurrent executors
                 // Do not do the cleanuptask in the current thread as we will unncessarily add to execution time
-                this.lastParallelRunCleanUpTask = Task.Run(() =>
+                Task.Run(() =>
```

Comment:
There doesn't seem any benefit of creating a Task and Waiting on it synchronously. Same comment for discovery logic.

Reply:
How does an exception during cleanup affect the parallel logic?

Reply:
The only place exception can be thrown is while Closing testhost(we also handle this under try catch)), so this means that testhost did not close(possibly), in such cases we should not be sending discovery/execution complete raw message anyways.